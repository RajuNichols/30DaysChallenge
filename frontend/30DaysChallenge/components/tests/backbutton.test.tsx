import React from "react";
import { TouchableOpacity } from "react-native";
import { Ionicons } from "@expo/vector-icons";
import renderer from "react-test-renderer";

import BackButton from "../backbutton";

describe("BackButton", () => {
  const navigation = { goBack: jest.fn() };

  it("renders correctly", () => {
    const tree = renderer.create(<BackButton navigation={navigation} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("calls navigation.goBack when pressed", () => {
    const wrapper = renderer.create(<BackButton navigation={navigation} />);
    const button = wrapper.root.findByType(TouchableOpacity);

    button.props.onPress();
    expect(navigation.goBack).toHaveBeenCalledTimes(1);
  });

  it("displays the correct icon", () => {
    const wrapper = renderer.create(<BackButton navigation={navigation} />);
    const icon = wrapper.root.findByType(Ionicons);

    expect(icon.props.name).toEqual("arrow-back");
    expect(icon.props.size).toEqual(24);
    expect(icon.props.color).toEqual("black");
  });
});