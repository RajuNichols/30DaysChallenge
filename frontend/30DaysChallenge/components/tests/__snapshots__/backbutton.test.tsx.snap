// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "display": "flex",
      "flexDirection": "row",
      "left": "20%",
      "opacity": 1,
      "top": "15%",
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "fontFamily": "Inter_800ExtraBold",
        "top": 2,
      }
    }
  >
    Back
  </Text>
</View>
`;
